packages/fxa-content-server/app/scripts/models/account.js

```diff
@@ -293,6 +295,37 @@ const Account = Backbone.Model.extend(
       );
     },
 
+    /**
+     * Given a user's `kB` and a pipeline public key chosen at random from the
+     * key bundle, derive the ecosystem_user_id and encrypt it with the
+     * pipeline key, yielding the ecosystem_anon_id. Return the
+     * ecosystem_anon_id.
+     *
+     * @param {String} kB - user's kB
+     * @param {String} pipelineKey - a randomly-chosen pipeline public key
+     * in JWK format
+     * @returns {Promise} A promise that will be fulfilled with the user's
+     * ecosystem_anon_id in JWK format
+     */
+    async generateEcosystemAnonID(kB, pipelineKey) {
+      const uid = this.get('uid');
+      const keyData = {
+        // This identifier comes from the 'Account Ecosysten Telemetry
+        // Identifiers' google doc.
+        identifier: 'https://identity.mozilla.com/ids/ecosystem_telemetry',
+        // Following fxa-auth-server/lib/oauth/routes/key_data.js#L14-L20, FxA
+        // doesn't yet support per-user scoped key rotation; use a block of
+        // zeros instead.
+        keyRotationSecret:
+          '0000000000000000000000000000000000000000000000000000000000000000',
+        keyRotationTimestamp: 1595287165358,
+      };
+
+      const ecosystemUserId = await _deriveScopedKeys(kB, uid, keyData);
```

Comment:
I expect that `derivedScopedKeys` here will be returning a full JWK object with `kid`, `crv` etc properties. You want *just* the `k` property which is the underlying key bytes. (Also I think the `k` property is a base64 string in this case, but the data pipeline wants it as a hex string).

Reply:
> data pipeline wants it as a hex string

Oh, I thought it was base64 everywhere. @6a68 do we need to do any coordination with profile-server/auth-server to ensure we are saving in correct format?

Reply:
Good catch, @rfk! And good question @vbudhram, I believe we treat it as an opaque string without enforcing any particular encoding, so I think it should be encoding agnostic--but we should double check.

Reply:
Following up on this base64/hex issue, we use the JWK compact serialization everywhere, which is urlsafe base64. The server endpoints just treat it as an opaque string. I've called out the need for conversion to hex in the data pipeline submission issue (#5998).

Reply:
To be clear, I'm talking specifically about the value *inside* the JWE here (that is, the `eco_uid`) which is not affected by the choice of encoding or formatting or the JWE wrapper. The data pipeline will decrypt the JWE and validate the contained value, rejecting any values that are not [a 64-character hex string](https://github.com/mozilla-services/mozilla-pipeline-schemas/blob/490cfe00c401e079277c00715d8dda55b7cd81cd/schemas/firefox-accounts/account-ecosystem/account-ecosystem.1.schema.json#L21).

Reply:
To be even more explicit, I expect this code will need to look something like:

```
ecosystemUserJWK = await _deriveScopedKeys(kB, uid, keyData);
ecosystemUserId = encode_to_hex(decode_from_base64(ecosystemUserJWK.k));
ecosystemAnonId = await encrypt(ecosystemUserId, pipelineKey);
```

(Alternately we could decide that we want `ecosystemUserId` to be a base64 string everywhere and update the pipeline schema to match, but hex identifiers seem more common in other parts of telemetry)

Reply:
Oh, of course, thanks for pointing that out @rfk. The value has to be encoded correctly _before_ it's encrypted using the pipeline key, since the pipeline will decrypt and then need to validate what it's decrypted.

I'll update my patch accordingly...and drop the incorrect comment from the metrics bug